code/scripting/api/objs/debriefing.cpp

```diff
@@ -0,0 +1,132 @@
+#include "debriefing.h"
+
+namespace scripting {
+namespace api {
+
+debrief_stage_h::debrief_stage_h() {}
+debrief_stage_h::debrief_stage_h(debrief_stage* db_stage) : stage(db_stage) {}
+debrief_stage* debrief_stage_h::getStage() const
+{
+	return stage;
+};
+
+bool debrief_stage_h::IsValid() const
+{
+	return stage != nullptr;
+}
+
+//**********HANDLE: debriefing
+ADE_OBJ(l_DebriefStage, debrief_stage_h, "debriefing_stage", "Debriefing stage handle");
+
+ADE_VIRTVAR(Text, l_DebriefStage, nullptr, "The text of the stage", "string", "The text")
+{
+	debrief_stage_h stage;
+	if (!ade_get_args(L, "o", l_DebriefStage.Get(&stage))) {
+		return ADE_RETURN_NIL;
+	}
+
+	if (ADE_SETTING_VAR) {
+		LuaError(L, "This property is read only.");
+	}
+
+	return ade_set_args(L, "s", stage.getStage()->text);
+}
+
+ADE_VIRTVAR(AudioFilename,
+	l_DebriefStage,
+	nullptr,
+	"The filename of the audio file to play",
+	"string",
+	"The file name")
+{
+	debrief_stage_h stage;
+	if (!ade_get_args(L, "o", l_DebriefStage.Get(&stage))) {
+		return ADE_RETURN_NIL;
+	}
+
+	if (ADE_SETTING_VAR) {
+		LuaError(L, "This property is read only.");
+	}
+
+	return ade_set_args(L, "s", stage.getStage()->voice);
+}
+
+ADE_VIRTVAR(Recommendation, l_DebriefStage, nullptr, "The recommendation text of the stage", "string", "The recommendation text")
+{
+	debrief_stage_h stage;
+	if (!ade_get_args(L, "o", l_DebriefStage.Get(&stage))) {
+		return ADE_RETURN_NIL;
+	}
+
+	if (ADE_SETTING_VAR) {
+		LuaError(L, "This property is read only.");
+	}
+
+	return ade_set_args(L, "s", stage.getStage()->recommendation_text);
+}
+
+ADE_VIRTVAR(isVisible,
+	l_DebriefStage,
+	nullptr,
+	"The result of the stage formula",
+	"boolean",
+	"true if the stage should be displayed, false otherwise")
+{
+	debrief_stage_h stage;
+	if (!ade_get_args(L, "o", l_DebriefStage.Get(&stage))) {
+		return ADE_RETURN_NIL;
```

Comment:
This can be changed to return false.  Invalid stages certainly won't be shown, and there are other ways elsewhere to check for stage validity.